packages/profile-sync-controller/src/AuthenticationController.ts

```diff
@@ -0,0 +1,338 @@
+import type {
+  RestrictedControllerMessenger,
+  StateMetadata,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import type { InternalAccount } from '@metamask/keyring-api';
+import type { HandleSnapRequest } from '@metamask/snaps-controllers';
+
+import {
+  createSnapPublicKeyRequest,
+  createSnapSignMessageRequest,
+} from './AuthSnapRequests';
+import {
+  createLoginRawMessage,
+  getAccessToken,
+  getNonce,
+  login,
+} from './services/authentication-controller';
+import type { UserStorageControllerDisableProfileSyncing } from './UserStorageController';
+
+const THIRTY_MIN_MS = 1000 * 60 * 30;
+
+const controllerName = 'AuthenticationController';
+
+type SessionProfile = {
+  identifierId: string;
+  profileId: string;
+};
+
+type SessionData = {
+  /** profile - anonymous profile data for the given logged in user */
+  profile: SessionProfile;
+  /** accessToken - used to make requests authorized endpoints */
+  accessToken: string;
+  /** expiresIn - string date to determine if new access token is required  */
+  expiresIn: string;
+};
+
+type MetaMetricsAuth = {
+  getMetaMetricsId: () => string;
+};
+
+export type AuthenticationControllerState = {
+  /**
+   * Global isSignedIn state.
+   * Can be used to determine if "Profile Syncing" is enabled.
+   */
+  isSignedIn: boolean;
+  sessionData?: SessionData;
+  internalAccounts?: {
+    accounts: Record<string, InternalAccount>;
+    selectedAccount: string;
+  };
+};
+const defaultState: AuthenticationControllerState = { isSignedIn: false };
```

Comment:
Similar as for the notifications controller: We've been giving controllers a `getDefault<ControllerName>State` function rather than a variable:

``` suggestion
function getDefaultAuthenticationControllerState: AuthenticationControllerState {
  return { isSignedIn: false };
}
```

Reply:
addressed